resteasy-core/src/main/java/org/jboss/resteasy/plugins/delegates/MediaTypeHeaderDelegate.java

```diff
@@ -87,7 +88,7 @@ private static MediaType internalParse(String type) {
                 major = major.substring(0, paramIndex);
             }
             if (!MediaType.MEDIA_TYPE_WILDCARD.equals(major)) {
-                throw new IllegalArgumentException(Messages.MESSAGES.failureParsingMediaType(type));
+                throw new NotAcceptableException(Messages.MESSAGES.failureParsingMediaType(type));
```

Comment:
This isn't the correct place for this to be thrown. The java doc for `HeaderDelegate` states:
```quote
[IllegalArgumentException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/IllegalArgumentException.html) - if the supplied string cannot be parsed or is null.
```
Reference https://jakarta.ee/specifications/restful-ws/3.1/apidocs/jakarta.ws.rs/jakarta/ws/rs/ext/runtimedelegate.headerdelegate

Reply:
Hello @jamezp , thank you very much for the review.
I looked into it, and the place where the MediaTypeHeaderDelegate.fromString() is called is the MediaType.valueOf() method, which is from an external library jakarta.ws.rs-api.
Can I ask? What do you think I should do in such a case? Is this still an issue regarding the RESTEASY project?

Reply:
I think what we need to do is create a reproducer. The original issue is against RESTEasy Reactive which is part of Quarkus. Chunks of RESTEasy Reactive were pulled from RESTEasy.  I believe this is how it was fixed in Quarkus, https://github.com/quarkusio/quarkus/pull/34874. However, we need to attempt to reproduce it in RESTEasy, likely with WildFly, and then we can track down the best place to fix it.

Reply:
Thank you very much for the suggestion @jamezp ! :)
So, I created a reproducer with quickstart/spring-resteasy. I uploaded war into WildFly and used `curl -i -H "Accept: invalid" -X GET http://localhost:8080/spring-resteasy/basic`. From the server log https://pastebin.com/XFjCAn5Q, I discovered that MediaTypeHeaderDelegate.parse() could be the method for applying changes. I force-pushed new changes, but I had to change some tests. What do you think about that?

Reply:
I don't know what I thought when the parse method provided output for the fromString method. I apologize.
Yes sure. I'll send you the war via GChat because it can't be uploaded here. 
I uploaded the war to WildFly and then used `curl -i -H "Accept: invalid" -X GET [http://localhost:8080/spring-resteasy/basic`](http://localhost:8080/spring-resteasy/basic). Hopefully, it is the right reproducer for the issue.
But I was so fixed on details that I only now realized that the endpoint is not returning 500 as issue description claims, but 400. ServletContainerDispatcher catches the IllegalArgmuentException.service here: https://github.com/khermano/resteasy/blob/main/resteasy-core/src/main/java/org/jboss/resteasy/plugins/server/servlet/ServletContainerDispatcher.java#L205, so I am not sure if the issue is valid.

Reply:
Hello @jamezp ,
I force-pushed some changes, but as I mentioned in the previous comment, I think the issue is no longer valid because I discovered that using the "invalid" Accept header returns 400 even before any change is added.
What do you think about that?